### DISCUSSION TITLE
Prettify website

### DISCUSSION THREAD

**Discussion by ludgie** [labwc/labwc#1442] [Ideas]
I suggest the following:
- Add favicon (or even svg favicon if img/labwc.png is available as svg)
- Add meta information to the website head for better discovering (social share preview) (super unnecessary but could look well)

**Comment by ahesford** [comment]
I absolutely oppose the entire concept of "social share previews". This kind of SEO marketing nonsense has destroyed the classic wonder of the Internet. Open-source projects should stay away from "like and subscribe" prostitution and focus on building quality software.

**Reply by ludgie** [reply]
Understandable the reply to the second suggestion
Regarding the first, it's acceptable, right?

**Reply by redtide** [reply]
He already stated it as "super unnecessary" 😄 
The meaning of this idea is in general to add a favicon and meta information in the head section, which IMO seems reasonable (though IDK how is made currently).

**Reply by ludgie** [reply]
Regarding the "SEO marketing nonsense", was meaning just add the meta tags that the OpenGraph protocol specifies, which is supported by Mastodon, Facebook, LinkedIn, Whatsapp, and a large etc. (Twitter has its own ones, but I think supporting it doesn't make sense seeing the current status of the platform.)

Further info:
https://ogp.me/
https://www.opengraph.xyz/
https://docs.joinmastodon.org/entities/PreviewCard/

for example, currently the project website without opengraph tags looks like this: https://socialsharepreview.com/?url=https://labwc.github.io

Also, I'd submit a PR adding this if the maintainers are interested

**Reply by ahesford** [reply]
My biggest problem with OpenGraph is that some of it (`og:title`, `og:url`, `og:description` and others) needlessly duplicates existing standard META tags, and a lot of others (*e.g.*, image metadata like format and size) just exist to optimize machine handling at the expense of injecting a human to maintain this stuff.

**Reply by Consolatis** [reply]
Well, labwc.github.io is created by a script anyway. So we can just throw some generic `og:` tags into `src/template_pre.html` and be done with it.

**Comment by Consolatis** [comment]
I do not consider adding some meta tags with description and logo + a favicon as SEO marketing nonsense.
So I am in favor of both.

**Reply by ahesford** [reply]
I agree that adding some meta tags offering a description and a favicon are fine, but I interpreted "social share previews" as falling down a rabbit-hole of custom HTML tags designed to promote Facebook/Twitter/Pinterest sharing as [described here](https://blog.logrocket.com/open-graph-sharable-social-media-previews/). That is the kind of alteration I find unappealing.

**Reply by ludgie** [reply]
Well, this is what I've been meaning. Though maybe the "social share previews" name isn't that appealing, in the end

**Reply by ahesford** [reply]
I'd rather see [standard meta tags](https://www.w3schools.com/tags/tag_meta.asp).

**Reply by ludgie** [reply]
I can add that too to the PR

**Reply by Consolatis** [reply]
IMHO the `og:` tags are fine.
For the `twitter:` (or any other company) prefix I agree that we should \*not\* support them.

**Comment by johanmalm** [comment]
Looks like I joined the party to late to take part in this discussion. Have already merged the additional tags.... hope that's okay.
I suspect we're pretty aligned: Stay clear of stars/likes/etc. No java-script. Make it easy for people to work offline, on modest hardware, etc.
One of the reasons that we setup the website the way we did is that you can just clone it and read the `src/*.md` files offline if you want. Or run the build-script and read it in a brower.

**Comment by ludgie** [comment]
Closing as the suggestions have been implemented with https://github.com/labwc/labwc.github.io/pull/40 and https://github.com/labwc/labwc.github.io/pull/41

**Comment by ludgie** [comment]
Reopening, just adding that the image looks super cursed 😂
![image](https://github.com/labwc/labwc/assets/77110723/f89db7c7-3961-4234-bf07-be3e61ec4f66)

If I can get the font being used in labwc2-with-text.png I could generate a 1200x630 image and fix that

**Reply by Consolatis** [reply]
Just slightly cursed :)

Replied in https://github.com/labwc/labwc.github.io/pull/40

**Reply by ludgie** [reply]
Proof of concept
![text1](https://github.com/labwc/labwc/assets/77110723/013e45bf-f5f8-4495-8611-39e9b92677e3)

Uses liberation mono
![image](https://github.com/labwc/labwc/assets/77110723/fb4975a4-1ca0-4182-bf4b-43805acf8c2e)

(the background colors are only so I can distinguish the separations)

**Reply by ludgie** [reply]
I'm open for feedback in case I can't get the font of the original labwc-with-text image made by @01micko (I think), and so replace that one with this one by providing the svg and exported png

**Reply by Consolatis** [reply]
Looks good to me. Is it using a transparent background? If yes the top side of the cube may be white when rendered on a white background, maybe it would be good to have a solid black background instead.

**Reply by johanmalm** [reply]
Sorry - can't keep up. I have the original `labwc2-with-text.svg`
Didn't realise it wasn't in the repo.
Will push later.

**Reply by ludgie** [reply]
No worries, take your time please, I don't want to be an annoyance to anyone, and also, thanks

**Reply by johanmalm** [reply]
Have pushed it :smile:
The font was cantarell.
